netboot-oci-artifacts.md

```diff
@@ -0,0 +1,96 @@
+currently it has MIME type `application/vnd.unknown.artifact.v1`.
+
+## Image layers
+
+Each layer in the manifest represents provided at build time PXE files:
+
+* vmlinuz
+* initrd.img
+* shim.efi
+* grubx64.efi
+
+Each layer must have:
+
+* the MIME type `application/x-netboot-file` or `application/x-netboot-file+zstd` if compressed
+* the annotations `org.opencontainers.image.description` and `org.opencontainers.image.title`
+  describing each file
+
+If the layer is not stored in a registry, then the optional `urls` property of the layer descriptor
+can be used to point to the layer contents.
+
+
+For the rest of image definition refer to the OCI specs.
+
+
+## Example OCI Image Manifest containing PXE files stored as OCI artifacts
+
+``` json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.unknown.artifact.v1",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+    "size": 0,
+    "annotations": {
+      "org.opencontainers.image.description": "Anaconda installer bootloader and Linux kernel/initrd. Usage: oras pull ghcr.io/lzap/bootc-netboot-example:TAG"
+    }
+  },
+  "layers": [
+    {
+      "mediaType": "application/x-netboot-file",
+      "digest": "sha256:0d7a9a3c4804334b23cd43ffc3aedad4620192d9c520e2f466f56b96aeb2a284",
+      "size": 13335480,
+      "annotations": {
+        "org.opencontainers.image.description": "Anaconda installer Linux kernel",
+        "org.opencontainers.image.title": "vmlinuz"
+      }
+    },
+    {
+      "mediaType": "application/x-netboot-file",
+      "digest": "sha256:4080a4d952d5145625d18b822214982a87ad981c254fcac671ca9ea245da5e3d",
+      "size": 102417772,
+      "annotations": {
+        "org.opencontainers.image.description": "Anaconda installer init RAM disk",
+        "org.opencontainers.image.title": "initrd.img"
+      }
+    },
+    {
+      "mediaType": "application/x-netboot-file",
+      "digest": "sha256:32e77976ebbc915f77dd7f15d66a52cb177d5a9d2ee1794b173390b67495c047",
+      "size": 946736,
+      "annotations": {
+        "org.opencontainers.image.description": "SecureBoot shim EFI binary",
+        "org.opencontainers.image.title": "shim.efi"
+      }
+    },
+    {
+      "mediaType": "application/x-netboot-file",
+      "digest": "sha256:735284626212a6267c0e90dab2428e8f82c182af17aec567c80838d219d9fa42",
+      "size": 2532984,
+      "annotations": {
+        "org.opencontainers.image.description": "Grub bootloader binary",
+        "org.opencontainers.image.title": "grubx64.efi"
+      }
+    }
+  ],
+  "annotations": {
+    "org.opencontainers.image.created": "2024-02-06T17:18:04Z",
+    "org.opencontainers.image.description": "Anaconda installer bootloader and Linux kernel/initrd. Usage: oras pull ghcr.io/lzap/bootc-netboot-example:TAG"
+  }
+}
+```
```

Comment:
Can you also add `install.img` (stage two image)? I initially forget it during our testing, it is required in order to boot into Anaconda (without kickstart tree).

I would like to propose adding an annotation called "entrypoint" with image title which is supposed to be the filename that is actually booted into. In this example it would be `shim.efi`. On the other hand, I specifically comment above we should not store any kind of configuration in the registry, which includes:

* `grub.cfg`
* kickstart

We could, however, actually include a configuration that would cause the installer to enter interactive mode. That would be something like (`grub.cfg`):

```
if [ ${grub_platform} == "pc" ]; then
  linux vmlinuz
  initrd initrd.img
else
  linuxefi vmlinuz
  initrdefi initrd.img
fi
```